tests/webpack-test/cases/loaders/cjs-loader-type/index.js

```diff
@@ -0,0 +1,15 @@
+it("should pass package.json type to loader", function () {
+	expect(require("cjs/loader.js!")).toBe("commonjs");
+	expect(require("./loader.js!")).toBe("undefined");
+});
+
+it("should pass 'commonjs' type to loader for .cjs", function () {
+	expect(require("cjs/loader.cjs!")).toBe("commonjs");
+	expect(require("./loader.cjs!")).toBe("commonjs");
+	// ORIGINAL WEBPACK COMMENT: TODO need fix in v8 https://github.com/nodejs/node/issues/35889
+	// ORIGINAL WEBPACK COMMENT: TODO otherwise this test case cause segment fault
+	// Turned on this as rspack checks extensions for loader type.
+	// So this will not fall into dynamic import which causes segment fault.
+	// See: crates/rspack_binding_options/src/plugins/js_loader/resolver.rs
+	expect(require("esm/loader.cjs!")).toBe("commonjs");
```

Comment:
Turned on a case that is failed in webpack. As we have a different strategy on loader type matching.

Reply:
This is the test case for the CI issue: https://github.com/web-infra-dev/rspack-ecosystem-ci/actions/runs/10521239196/job/29151639120#step:6:805

Reply:
In rsbuild, they relied on `type: "module"` and loader extension `.cjs` to correctly determine loader types.